db/migrate/20210401192727_add_salted_code_fingerprint_salt_cost_to_backup_code_configurations.rb

```diff
@@ -0,0 +1,7 @@
+class AddSaltedCodeFingerprintSaltCostToBackupCodeConfigurations < ActiveRecord::Migration[6.1]
+  def change
+    add_column :backup_code_configurations, :salted_code_fingerprint, :string, limit: 255
```

Comment:
What do you think about using a text column instead for this one? The others seem unlikely to grow beyond 255, but if we ever need to increase the size of the hash it may be nice to have the flexibility in the column

Reply:
my understanding is that `text` and `blob` are stored on disk separately, so that accessing them is essentially a `JOIN` every time, so I figured that since performance matters here, it would be better to keep these local?

Reply:
Yes I agree with @zachmargolis .  Keep them string.

Reply:
for the character types (varchar, char, text), the postgres docs say ([link](https://www.postgresql.org/docs/9.6/datatype-character.html)):

>There is no performance difference among these three types, apart from increased storage space when using the blank-padded type, and a few extra CPU cycles to check the length when storing into a length-constrained column.



I don't feel super strongly, but I'd default to text going forward unless there's a super concrete reason to use char/varchar.

Reply:
that's great to know! Maybe it's a MySQL thing... will fix. thanks for the link!

Reply:
fixed: 3908063ac0c7c4f35229952f3965ee3a94d7cada